partiql-tests-data/eval/primitives/path.ion

```diff
@@ -131,161 +130,230 @@ envs::{
 
 path::[
   {
-    name:"pathSimpleDotOnly",
-    statement:"b.c",
-    assert:{
-      evalMode:[EvalModeCoerce, EvalModeError],
-      result:EvaluationSuccess,
-      output:100
+    name: "pathSimpleDotOnly",
+    statement: "b.c",
+    assert: {
+      evalMode: [
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      result: EvaluationSuccess,
+      output: 100
     }
   },
   {
-    name:"pathDotOnly",
-    statement:"a.b.c.d.e",
-    assert:{
-      evalMode:[EvalModeCoerce, EvalModeError],
-      result:EvaluationSuccess,
-      output:5
+    name: "pathDotOnly",
+    statement: "a.b.c.d.e",
+    assert: {
+      evalMode: [
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      result: EvaluationSuccess,
+      output: 5
     }
   },
   {
-    name:"pathDotMissingAttribute",
-    statement:"a.z IS MISSING",
-    assert:{
-      evalMode:[EvalModeCoerce, EvalModeError],
-      result:EvaluationSuccess,
-      output:true
-    }
+    name: "pathDotMissingAttribute",
+    statement: "a.z IS MISSING",
+    assert: [
+      {
+        evalMode: EvalModeCoerce,
+        result: EvaluationSuccess,
+        output: true
+      },
+      {
+        evalMode: EvalModeError,
+        result: EvaluationFail
+      }
+    ]
   },
   {
-    name:"pathMissingDotName",
-    statement:"(MISSING).a IS MISSING",
-    assert:{
-      evalMode:[EvalModeCoerce, EvalModeError],
-      result:EvaluationSuccess,
-      output:true
-    }
+    name: "pathMissingDotName",
+    statement: "(MISSING).a IS MISSING",
+    assert: [
+      {
+        evalMode: EvalModeCoerce,
+        result: EvaluationSuccess,
+        output: true
+      },
+      {
+        evalMode: EvalModeError,
+        result: EvaluationFail
+      }
+    ]
   },
+  // TODO: LOOK AT THIS? Should this be NULL?
   {
-    name:"pathNullDotName",
-    statement:"(NULL).a IS MISSING",
-    assert:{
-      evalMode:[EvalModeCoerce, EvalModeError],
-      result:EvaluationSuccess,
-      output:true
+    name: "pathNullDotName",
+    statement: "(NULL).a IS MISSING",
+    assert: {
+      evalMode: [
+        EvalModeCoerce,
+        EvalModeError
+      ],
```

Comment:
This is really weird...

Build up on the previous test cases: 

If MISSING.a is MISSING failed evalMode: EvalModeError, why should NULL.a succeed? 

I understand you can coerce NULL to any type, but it does seem weird to me ....

Reply:
Nonblocking: but we can think of path navigation as a function that is not nullCall?